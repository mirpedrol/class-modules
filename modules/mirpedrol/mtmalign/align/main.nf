process MTMALIGN_ALIGN {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-5bcf71dc66dac33d8e003c5e78043b80f5c7f269:8f0e486d46f7ab38892c1a8f78d2894a549d03b5-0':
        'biocontainers/mulled-v2-5bcf71dc66dac33d8e003c5e78043b80f5c7f269:8f0e486d46f7ab38892c1a8f78d2894a549d03b5-0' }"

    input:
    tuple val(meta), path(pdbs)

    output:
    tuple val(meta), path("${prefix}.aln.gz"), emit: alignment
    path("versions.yml")                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    # decompress input files if required
    if ls ./*.pdb.gz 2&> /dev/null; then
        unpigz -d ./*.pdb.gz
    fi

    # construct input file for mtmalign
    ls *.pdb | sed s/\\ /\\n/ > input_list.txt

    mtm-align -i input_list.txt
    mv ./mTM_result/result.fasta ${prefix}.aln

    # Remove ".pdb" from the ids in the alignment file
    sed -i 's/\\.pdb//g' ${prefix}.aln

    # compress output file
    pigz -p ${task.cpus} ${prefix}.aln

    # mtm-align -v prints the wrong version 20180725, so extract it from the cosmetic output in the help message
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mTM-align: \$( mtm-align -h | grep -e "\\(Version [[:digit:]]*\\)" | grep -oe "[[:digit:]]*" )
        pigz: \$(echo \$(pigz --version 2>&1) | sed 's/^.*pigz\\w*//' ))
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.aln
    pigz ${prefix}.aln

    # mtm-align -v prints the wrong version 20180725, so extract it from the cosmetic output in the help message
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mTM-align: \$( mtm-align -h | grep -e "\\(Version [[:digit:]]*\\)" | grep -oe "[[:digit:]]*" )
        pigz: \$(echo \$(pigz --version 2>&1) | sed 's/^.*pigz\\w*//' ))
    END_VERSIONS
    """
}
