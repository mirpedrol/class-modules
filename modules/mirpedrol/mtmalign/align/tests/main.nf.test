nextflow_process {

    name "Test Process MTMALIGN_ALIGN"
    script "../main.nf"
    process "MTMALIGN_ALIGN"

    tag "modules"
    tag "modules_mirpedrol"
    tag "mtmalign"
    tag "mtmalign/align"

    setup {

        run("UNTAR") {
            script "../../../../../modules/mirpedrol/untar/main.nf"
                process {
                    """
                        archive = file("https://raw.githubusercontent.com/nf-core/test-datasets/multiplesequencealign/testdata/structures/seatoxin-ref.tar.gz", checkIfExists: true)
                        input[0] = Channel.of(tuple([id:'test'], archive))
                    """
                }
        }
    }

    test("seatoxin - aln") {

        when {
            process {
                """
                    input[0] = UNTAR.out.untar.map { meta,dir -> [[ id:'test' ], file(dir).listFiles().collect()]}
                """
            }
        }

        then {
            assertAll(
                { assert process.success  },
                // mTMalign may be nondeterministic, just check if the pdbs are all in there
                { assert path(process.out.alignment[0][1]).getTextGzip().contains(">1ahl") },
                { assert path(process.out.alignment[0][1]).getTextGzip().contains(">1apf") },
                { assert path(process.out.alignment[0][1]).getTextGzip().contains(">1atx") },
                { assert path(process.out.alignment[0][1]).getTextGzip().contains(">1bds") },
                { assert path(process.out.alignment[0][1]).getTextGzip().contains(">1sh1") },
                { assert snapshot(process.out.versions).match("versions1") }
            )
        }
    }

    test("seatoxin - stub") {

        options '-stub'

        when {
            process {
                """
                    input[0] = UNTAR.out.untar.map { meta,dir -> [[ id:'test' ], file(dir).listFiles().collect()]}
                """
            }
        }

        then {
            assertAll(
                { assert process.success  },
                { assert snapshot(process.out).match() }
            )
        }
    }

}