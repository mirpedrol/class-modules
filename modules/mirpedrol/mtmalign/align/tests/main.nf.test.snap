{
    "versions1": {
        "content": [
            [
                "versions.yml:md5,7cbacec15bb9e0c8cbb27610bde74c10"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-03T11:01:37.28539854"
    },
    "seatoxin - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "/Users/jmir/Work/class-modules/.nf-test/tests/4fe3d19c96a23d526512dafafb38acbc/work/5f/8b829def671cadf4289d4321710bc1/test.aln.gz"
                    ]
                ],
                "1": [
                    "versions.yml:md5,7cbacec15bb9e0c8cbb27610bde74c10"
                ],
                "alignment": [
                    [
                        {
                            "id": "test"
                        },
                        "/Users/jmir/Work/class-modules/.nf-test/tests/4fe3d19c96a23d526512dafafb38acbc/work/5f/8b829def671cadf4289d4321710bc1/test.aln.gz"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7cbacec15bb9e0c8cbb27610bde74c10"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2025-01-30T15:10:15.032184"
    }
}